tests/test_commit_message.py

```diff
@@ -36,6 +36,58 @@ def test_commit_message_reviewers_replaced():
     assert commit_message == (FIRST_LINE, COMMIT_MESSAGE)
 
 
+@pytest.mark.parametrize(
+    "reviewers",
+    [
+        ["a"],
+        ["aa"],
+        [".a"],
+        ["..a"],
+        ["...a"],
+        ["a...a"],
+        ["a.b"],
+        ["a.b.c"],
+        ["-.-.-"],
+        ["a", "aa", ".a", "..a", "...a", "a...a", "a.b", "a.b.c", "-.-.-"],
+    ],
+)
+def test_various_reviewer_name_combinations(reviewers):
+    """Tests single and multiple characters separated by various combinations
+    of periods for single and multiple reviewers.
+
+    Args:
+        - reviewers (list of str): a parameterized list of reviewer names
+    """
+    commit_message = format_commit_message(
+        "A title!", 1, reviewers, "", "http://phabricator.test/D123"
+    )
+
+    assert commit_message[0] == "Bug 1 - A title! r={}".format(",".join(reviewers))
+
+
+def test_period_at_end_of_reviewer_name():
+    """Tests that a period at the end of a reviewer's name does not get
+    parsed.
```

Comment:
This test doesn't actually end up parsing a reviewer which has a period at the end of the name. When you're calling...
```Python
    commit_message = format_commit_message(
        "A title! r=SDfsdfsd", 1, ["test.", "test2"], "", "http://phabricator.test/D123"
    )
```
...the third argument where you've added the the `.` is part of the list of reviewers to be written in, and isn't parsed at all. If you'd like to test the behaviour of parsing a reviewer with a trailing period it would have to be part of the first argument, which is the current title to be rewritten.

Currently, only the updates you made to `test_group_reviewers_replaced` parse reviewers that contain a period at all, the other changes just cause the formatter to write out reviewer names with periods.

Reply:
I removed this test and updated the other existing tests to be inline with this.